### PR TITLE
Use unique common name for edged-ca cert when apply'ing super-config.

### PR DESCRIPTION
**Pull request by arsing**
This is needed so that multiple Edge devices' edge CA certs in the same PEM
are distinguishable, such as when using Nested Edge.

Also remove hostname check from iotedge-check since it's redundant with
the check in aziotctl-check.

### CODE REVIEW

**Review by gordonwang0 — CHANGES_REQUESTED** [review]

File: edgelet/Cargo.toml [hunk 1]
```diff
@@ -15,3 +15,49 @@ members = [
     "management",
     "systemd",
 ]
+
+[patch."https://github.com/Azure/iot-identity-service"]
+aziot-cert-client-async = { path = "/home/arnavion/src/iot-identity-service/cert/aziot-cert-client-async" }
```
[the remark this answers is not in the document]
Comment (arsing) [reply edgelet/Cargo.toml:20]: That's why there's a TODO for it in the PR description.

**Review by huguesBouvier — APPROVED** [review]

**Review by daprilik — DISMISSED** [review]

**Review by massand — COMMENTED** [review]
The test-files changes have been done in PR #4556 already, along with the appropriate iotedge logic to use that for setting `auto_reprovision_mode`. I will merge that as soon as you approve so you rebase your PR.

**Comment by arsing** [comment]
>The test-files changes have been done in PR #4556 already, along with the appropriate iotedge logic to use that for setting `auto_reprovision_mode`. I will merge that as soon as you approve so you rebase your PR.

Yes, if yours goes in first, I'll update this one as necessary.

**Comment by arsing** [comment]
>>The test-files changes have been done in PR #4556 already, along with the appropriate iotedge logic to use that for setting `auto_reprovision_mode`. I will merge that as soon as you approve so you rebase your PR.
>
>Yes, if yours goes in first, I'll update this one as necessary.

Done.

**Review by massand — APPROVED** [review]

**Review by gordonwang0 — APPROVED** [review]

**Pull request merged**